test/integrations/testUtils.ts

```diff
@@ -364,3 +376,123 @@ export const compareObjects = (obj1, obj2, logPrefix = '', differences: string[]
+  let metadata: ProxyMetdata = {
+    jobId: 1,
+    attemptNum: 1,
+    userId: 'default-userId',
+    destinationId: 'default-destinationId',
+    workspaceId: 'default-workspaceId',
+    sourceId: 'default-sourceId',
+    secret: {
+      accessToken: 'default-accessToken',
+    },
+    dontBatch: false,
+  };
+  if (metadataInput) {
+    metadata = metadataInput;
+  }
+  const payload = {
+    version: 'v0',
+    type: 'REST',
+    userId: payloadParameters.userId || 'default-userId',
+    method: payloadParameters.method || 'POST',
+    endpoint: payloadParameters.endpoint || '',
+    headers: payloadParameters.headers || {},
+    params: payloadParameters.params || {},
+    body: {
+      JSON: payloadParameters.JSON || {},
+      JSON_ARRAY: payloadParameters.JSON_ARRAY || {},
+      XML: payloadParameters.XML || {},
+      FORM: payloadParameters.FORM || {},
+    },
+    files: payloadParameters.files || {},
+    metadata,
+  };
+  return removeUndefinedAndNullValues(payload);
+};
+
+export const generateProxyV1Payload = (
+  payloadParameters: any,
+  metadataInput?: ProxyMetdata[],
+  destinationConfig?: any,
+) => {
+  let metadata: ProxyMetdata[] = [
+    {
+      jobId: 1,
+      attemptNum: 1,
+      userId: 'default-userId',
+      destinationId: 'default-destinationId',
+      workspaceId: 'default-workspaceId',
+      sourceId: 'default-sourceId',
+      secret: {
+        accessToken: 'default-accessToken',
+      },
+      dontBatch: false,
+    },
+  ];
+  if (metadataInput) {
+    metadata = metadataInput;
+  }
+  const payload = {
+    version: 'v1',
+    type: 'REST',
+    userId: payloadParameters.userId || 'default-userId',
+    method: payloadParameters.method || 'POST',
+    endpoint: payloadParameters.endpoint || '',
+    headers: payloadParameters.headers || {},
+    params: payloadParameters.params || {},
+    body: {
+      JSON: payloadParameters.JSON || {},
+      JSON_ARRAY: payloadParameters.JSON_ARRAY || {},
+      XML: payloadParameters.XML || {},
+      FORM: payloadParameters.FORM || {},
+    },
+    files: payloadParameters.files || {},
+    metadata,
+    destinationConfig: destinationConfig || {},
+  };
+  return removeUndefinedAndNullValues(payload);
+};
+
+// -----------------------------
+// Zod validations
+
+export const validateTestWithZOD = (testPayload: TestCaseData, response: any) => {
+  // Validate the resquest payload
+  switch (testPayload.feature) {
+    // case 'router':
+    //   RouterSchema.parse(responseBody);
+    //   break;
+    // case 'batch':
+    //   BatchScheam.parse(responseBody);
+    //   break;
+    // case 'user_deletion':
+    //   DeletionSchema.parse(responseBody);
+    //   break;
+    // case 'processor':
+    //   ProcessorSchema.parse(responseBody);
+    //   break;
+    case 'dataDelivery':
+      if (testPayload.version === 'v0') {
+        ProxyV0RequestSchema.parse(testPayload.input.request.body);
+        if (testPayload.scenario === 'Oauth') {
```

Comment:
this should be applied based destination config or something but not based on testPayload

Reply:
Initially we don't have destination config as a request parameter for the Proxy request and hence it is made option in test payload generation https://github.com/rudderlabs/rudder-transformer/blob/7ce0a6605ce9a65d29f69d4be5f1c54f382ab12c/test/integrations/testUtils.ts#L418.

Hence we went with scenario based segregation and accordingly applied validation.

If we want to do this we have to make `destiantionConfig` mandatory and as part of test payload generation we have to always ask test writer to add it

Reply:
this is part of the destination definition so, can't we keep some destination configuration in the transformer or have the integration config as a dependency and do it

Reply:
I would say its a bit tricky to add validations at this moment. For now we can enforce the process but we can figure out the automated validations as part of next step